examples/late.rs

```diff
@@ -10,23 +10,20 @@ use panic_semihosting as _;
 #[rtic::app(device = lm3s6965)]
 mod app {
     use cortex_m_semihosting::{debug, hprintln};
-    use heapless::{
-        consts::*,
-        i,
-        spsc::{Consumer, Producer, Queue},
-    };
+    use heapless::spsc::{Consumer, Producer, Queue};
     use lm3s6965::Interrupt;
 
     // Late resources
     #[resources]
     struct Resources {
-        p: Producer<'static, u32, U4>,
-        c: Consumer<'static, u32, U4>,
+        p: Producer<'static, u32, 5>,
+        c: Consumer<'static, u32, 5>,
     }
 
     #[init]
     fn init(_: init::Context) -> (init::LateResources, init::Monotonics) {
-        static mut Q: Queue<u32, U4> = Queue(i::Queue::new());
+        // NOTE: Queue must contain 1 advanced element to work fine
```

Comment:
I haven't read into things too much, but why is this the case? Is it something to do with the examples, or just a constant thing that has to be taken note of?

Reply:
https://docs.rs/heapless/0.7.1/heapless/spsc/struct.Queue.html

"A statically allocated single producer single consumer queue with a capacity of N - 1 elements"

since 0.7.0